plugins/npm/src/tasks/npm-publish.ts

```diff
@@ -1,52 +1,54 @@
+import { spawn } from 'node:child_process'
 import { ToolKitError } from '@dotcom-tool-kit/error'
 import { Task } from '@dotcom-tool-kit/types'
 import { semVerRegex, prereleaseRegex, releaseRegex } from '@dotcom-tool-kit/types/lib/npm'
-import pacote from 'pacote'
 import { readState } from '@dotcom-tool-kit/state'
-import pack from 'libnpmpack'
-import { publish } from 'libnpmpublish'
-import { styles } from '@dotcom-tool-kit/logger'
-import tar from 'tar'
```

Comment:
**suggestion**: imo this change is completely unrelated to the issue that's being fixed, and so should be done in its own PR. these libraries aren't "internal", they're their own public packages with a documented interface. i used the lower-level libraries rather than wrapping the CLI so we'd have more control over the error handling and logging. i'm not completely against this change, i just don't think it should be bundled with this bug fix.

Reply:
To clarify, is the change being referred to moving away from libraries such as `libnpmpublish` to using the npm CLI directly? If so, I am not sure how to split this away from the current PR. The code changes to move to using the CLI will be the same as fixing this bug. I can change the commit message to describe the change as moving away from `libnpm*` packages rather than fixing the bug?

I understand that `libnpmpack` and others are published public packages, my reference to them being internal is because the npm CLI internals use the packages.

Reply:
you could have fixed the bug separately via Ivo's suggestion of directly writing the version to `package.json`. moving away from these libraries is a much bigger change to how the task works, which makes me nervous.

Reply:
As discussed on slack - we are okay to merge this as is for now.

Reply:
> As discussed on slack - we are okay to merge this as is for now.

hey, in the future could we try and keep code review discussion within the PR itself rather than moving to Slack DMs? I acknowledge it's slightly harder to hold a back-and-forth discussion in a commenting format rather than an IRC format but I think it's important to be able to be able to easily refer back to why these decisions were made. as it stands, merging this PR which includes various unrelated changes appears to have resulted in many followup bug fix PRs being required, and it's frustrating that I don't have access to the rationale for why this was ultimately merged.